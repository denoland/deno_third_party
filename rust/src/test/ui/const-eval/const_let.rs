// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![feature(const_let)]

fn main() {}

struct FakeNeedsDrop;

impl Drop for FakeNeedsDrop {
    fn drop(&mut self) {}
}

// ok
const X: FakeNeedsDrop = { let x = FakeNeedsDrop; x };

// error
const Y: FakeNeedsDrop = { let mut x = FakeNeedsDrop; x = FakeNeedsDrop; x };
//~^ ERROR constant contains unimplemented expression type

// error
const Z: () = { let mut x = None; x = Some(FakeNeedsDrop); };
//~^ ERROR constant contains unimplemented expression type
