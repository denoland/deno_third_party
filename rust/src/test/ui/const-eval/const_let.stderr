error[E0019]: constant contains unimplemented expression type
  --> $DIR/const_let.rs:25:55
   |
LL | const Y: FakeNeedsDrop = { let mut x = FakeNeedsDrop; x = FakeNeedsDrop; x };
   |                                                       ^

error[E0019]: constant contains unimplemented expression type
  --> $DIR/const_let.rs:29:35
   |
LL | const Z: () = { let mut x = None; x = Some(FakeNeedsDrop); };
   |                                   ^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0019`.
